cpp/src/io/avro/avro_gpu.cu

```diff
@@ -307,42 +352,64 @@ __global__ void __launch_bounds__(num_warps * 32, 2)
   } else {
     schema = schema_g;
   }
+
   if (block_id < blocks.size() and threadIdx.x == 0) { *blk = blocks[block_id]; }
   __syncthreads();
   if (block_id >= blocks.size()) { return; }
-  cur_row        = blk->first_row;
-  rows_remaining = blk->num_rows;
+
   cur            = avro_data + blk->offset;
   end            = cur + blk->size;
-  while (rows_remaining > 0 && cur < end) {
+  first_row      = blk->first_row + blk->row_offset;
+  cur_row        = blk->row_offset;
+  end_row        = first_row + blk->num_rows;
+  rows_remaining = blk->num_rows;
```

Comment:
Please declare at the point of initialization (if possible). We're trying to move away from the C-style declaration at the top of each function so consistency with the current code is not recommended in this regard :)